Nested Comments & Reply System

SydorukDmytro opened the pull request:
# Summary
This PR improves the Blazor comments UI and interaction flow by introducing a structured nested comment system, reply mode, and redesigned layout.
### 1. Added support for two-nested comments
- Child comments now render under their parent with a visual indent (ml-6).

- Tree is supplied 
directly from the backend (root + children).

### 2. Redesigned comment layout
- Username, avatar, timestamp, and reply button are aligned in a single horizontal row.

- Timestamp is right-aligned using Justify="SpaceBetween".

- Comment text is placed below the header with consistent indentation.

### 3. Implemented Reply Mode

- Clicking Reply sets ParentId and displays “Replying to @username”.

- After sending, reply mode resets.

- Both root comments and replies use the same component logic.

### 4. Improved UX consistency
- Updated typography for better readability.

- Introduced spacing adjustments to create a cleaner layout.

- Child comments use a compact variant to differentiate from root comments.

Screens:

- <img width="1919" height="1079" alt="image" src="https://github.com/user-attachments/assets/ff0317bf-18f0-43ea-b03f-3bbd08adaea1" />
- <img width="1919" height="1078" alt="image" src="https://github.com/user-attachments/assets/6c7b6c4f-032a-4399-9647-6cd24424f9cf" />

Commit:
Nested Comments & Reply System

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDetailsDialog.razor:
```diff
@@ -3,6 +3,7 @@
 @using Lumix.Blazor.Services.IServices
 @inject IPhotoService PhotoService
+@inject ICommentService CommentService
 @inject ISnackbar Snackbar
```
Інжекти краще переносити в razor.cs як

        [Inject] public CommentService CommentService { get; set; } = default!;

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDetailsDialog.razor.cs:
```diff
@@ -29,29 +28,38 @@ protected override async Task OnInitializedAsync()
+        }
 
-            //API Request
+        void Cancel() => MudDialog.Cancel();
```
не юзається. + завжди краще ставити public / private

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDetailsDialog.razor.cs:
```diff
@@ -29,29 +28,38 @@ protected override async Task OnInitializedAsync()
+                Snackbar.Add($"Не вдалося додати коментар: {response.ErrorMessage}", Severity.Error);
+                return;
+            }
+            await LoadCommentsAsync();
```
чи треба саме лоадити заново після додавання? Думаю достатньо додати просто в Photo.Comments

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor:
```diff
@@ -123,21 +212,14 @@
+    .comment-time {
+        min-width: 80px;
+        text-align: right;
+    }
```
перенести стилі у site.css

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor.cs:
```diff
@@ -0,0 +1,57 @@
+        [Parameter] public UserProfileDto CurrentUser { get; set; }
+        [Parameter] public EventCallback<CommentRequest> OnAddComment { get; set; }
+
+        private string newComment = string.Empty;
```
краще юзати проперті, якщо ці змінні юзаються у розмітці

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor.cs:
```diff
@@ -0,0 +1,57 @@
+        {
+            var diff = DateTime.UtcNow - createdAt.ToUniversalTime();
+
+            if (diff.TotalMinutes < 1) return "щойно";
```
класне рішення

YuriLukashenko commented on Lumix.Blazor/Services/CommentService.cs:
```diff
@@ -0,0 +1,62 @@
+    {
+        private readonly HttpService _httpService;
+        private readonly ILogger<PhotoService> _logger;
+        private readonly string _baseUrl = "https://localhost:7231/api/comment";
```
винести base url https://localhost:7231 в аппсеттінгс. В сервіси лишити тільки route, api/comment

YuriLukashenko commented on Lumix.Blazor/Services/CommentService.cs:
```diff
@@ -0,0 +1,62 @@
+        {
+            try
+            {
+                using var content = new MultipartFormDataContent();
```
що дає використання MultipartFormDataContent?

Commit:
refactor: migrate comment posting to JSON and improve component structure

- Replaced multipart/form-data comment submission with JSON-based requests
- Updated CommentService to return CommentDto for immediate UI state updates
- Refactored PhotoDetailsDialog to inject newly created comments into the local DTO tree
- Removed unnecessary comment reloads after POST
- Improved nested reply handling and comment sorting logic

- Moved all dependency injections from .razor markup to the .razor.cs code-behind
- Moved inline component styles into site.css for better maintainability
- Replaced private fields with public properties across components
- Extracted BaseUrl into appsettings.json and adjusted services to use configuration-based URLs

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDetailsDialog.razor:
```diff
@@ -2,8 +2,6 @@
 @using Lumix.Blazor.Data.Photo
```
можна ремувнути юзінги тут

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDetailsDialog.razor:
```diff
@@ -42,13 +40,6 @@
-    }
-</style>
-
 @code {
```
можна теж прибрати пустий блок

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDetailsDialog.razor.cs:
```diff
@@ -11,8 +13,9 @@ public partial class PhotoDetailsDialog
+        [Inject] public ICommentService CommentService { get; set; } = default!;
+        [Inject] public ISnackbar Snackbar { get; set; } = default!;
 
         public PhotoDto Photo { get; set; }
```
якщо об'єкт може бути nullable, то зробити його як `PhotoDto?  `

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDetailsDialog.razor.cs:
```diff
@@ -11,8 +13,9 @@ public partial class PhotoDetailsDialog
+        [Inject] public ISnackbar Snackbar { get; set; } = default!;
 
         public PhotoDto Photo { get; set; }
         private bool isLoading = true;
```
приватне поле, з нижнім підкреслюванням, `_isLoading`

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor:
```diff
@@ -11,15 +12,22 @@
+
+            <MudStack Row="true" Spacing="2" AlignItems="AlignItems.Center">
+                <MudAvatar Size="Size.Medium">
+                    <MudImage Src="@Photo.Author.ProfilePictureUrl" Alt="@Photo.Author.Username" />
```
через Alt не працюватиме. Краще типу як

```
     <MudAvatar Color="Color.Primary" Size="Size.Large">
         @if (string.IsNullOrEmpty(avatarUrl))
         {
             @buyer.Name.GetInitials("B");
         }
         else
         {
             <MudImage Src="@avatarUrl" />
         }
     </MudAvatar>
```

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor:
```diff
@@ -88,56 +163,36 @@
-    }
-</style>
-
 @code {
```
теж прибрати @code та юзінги

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor.cs:
```diff
@@ -0,0 +1,57 @@
+        [Parameter] public UserProfileDto CurrentUser { get; set; }
+        [Parameter] public EventCallback<CommentRequest> OnAddComment { get; set; }
+
+        public String NewComment { get; set; } = string.Empty;
```
string з малої

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor.cs:
```diff
@@ -0,0 +1,57 @@
+            ReplyToUsername = string.Empty;
+        }
+
+        private async Task StartReply(CommentDto comment)
```
async не пишеться, коли всередині нема await

YuriLukashenko commented on Lumix.Blazor/Services/AuthService.cs:
```diff
@@ -25,11 +26,14 @@ public AuthService(
         HttpService httpService,
         ILogger<AuthService> logger,
         IJSRuntime jsRuntime,
+        ApiSettings settings,
```
простіше задавати як IOptions < ApiSettings > options, і потім брати options.Value;
але і так ок

YuriLukashenko commented on Lumix.Blazor/Lumix.Blazor.csproj:
```diff
@@ -8,6 +8,13 @@
     <DockerDefaultTargetOS>Linux</DockerDefaultTargetOS>
   </PropertyGroup>
 
+  <ItemGroup>
```
remove

YuriLukashenko commented on Lumix.Blazor/Program.cs:
```diff
@@ -49,10 +51,15 @@
     });
 
+builder.Services.Configure<ApiSettings>(builder.Configuration.GetSection("ApiSettings"));
+builder.Services.AddSingleton(resolver =>
```
use IOptions< ApiSettings > is constructors directly.

Commit:
Removed unused usings and code-blocks, renamed variables and properties, replaced using ApiSettings directly with using IOptions
fixed the display of the alternative photo in mud avatar

YuriLukashenko commented on Lumix.Blazor/Components/Photo/PhotoDialogComments.razor:
```diff
@@ -11,15 +10,31 @@
+                    @if (string.IsNullOrEmpty(Photo.Author.ProfilePictureUrl))
+                    {
+                        @Photo.Author.Username.GetInitials("Unknown")
+                        ;
```
можна без ;
